### PR TITLE
[Scenes] Fix Yaml 2_1 to 2_4 to enable on CI

### PR DESCRIPTION
**Pull request by lpbeliveau-silabs**
Makes the yaml 2_1 to 2_4 compliant with test plan so they can be automated on CI.

### CODE REVIEW

File: src/app/tests/suites/certification/Test_TC_S_2_2.yaml [hunk 1]
```diff
@@ -237,6 +237,8 @@ tests:
                 value: 0x0001
               - name: "SceneID"
                 value: 0x01
+      response:
+          error: ''
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_2_2.yaml:241]: This should list the error expected based on the spec, if this is supposed to error, no?

That said, per test plan this is not supposed to error, so why is this being added?

File: src/app/tests/suites/certification/Test_TC_S_2_2.yaml [hunk 2]
```diff
@@ -264,7 +266,7 @@ tests:
       response:
           value: 0x0001
           constraints:
-              type: groupid
+              type: int16u
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_2_2.yaml:269]: Again, why this change?  Per spec this should in fact be a group-id.  If our YAML processor is not handling that right, where is the issue with links to it from the YAML tracking that?

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml [hunk 3]
```diff
@@ -184,7 +180,7 @@ tests:
           "Step 2b: TH sends a AddScene command to DUT with the GroupID field
           set to G1, the SceneID field set to 0x01, the TransitionTime field set
           to 0x0001 (1s) and a set of extension fields appropriate to AC1."
-      PICS: S.S.C40.Rsp && PICS_SDK_CI_ONLY
+      PICS: (!S.S.C40.Rsp) && PICS_SDK_CI_ONLY
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_2_3.yaml:183]: This does not seem to match the test plan.

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml [hunk 4]
```diff
@@ -274,7 +266,7 @@ tests:
           [1670970505.890328][5742:5744] CHIP:TOO: Endpoint: 1 Cluster: 0x0000_0005 Command 0x0000_0040
           [1670970505.890458][5742:5744] CHIP:TOO:   EnhancedAddSceneResponse: {
           [1670970505.890551][5742:5744] CHIP:TOO:     status: 0
-          [1670970505.890608][5742:5744] CHIP:TOO:     groupId: 1
+          [1670970505.890608][5742:5744] CHIP:TOO:     groupId: 101
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_2_3.yaml:269]: It's 0x101 in the header above.  Will it really get logged as "101" here instead of 257?  Same below.

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml [hunk 5]
```diff
@@ -389,18 +381,14 @@ tests:
...
-                                    },
                                 ],
                         },
                     ]
 
     - label:
           "Step 3b: TH sends a ViewScene command to DUT with the GroupID field
           set to G1 and the SceneID field set to 0x01."
-      PICS: S.S.C41.Rsp && PICS_SDK_CI_ONLY
+      PICS: (!S.S.C41.Rsp) && PICS_SDK_CI_ONLY
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_2_3.yaml:391]: Neither the old nor the new PICS seems to match the test plan?  But the test plan also seems pretty weird, in that it does EnhancedViewScene and then does ViewScene for the same exact scene....

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml [hunk 6]
```diff
@@ -558,7 +542,7 @@ tests:
           field set to G1 and the scene ID field set to 0x01."
       PICS: S.S.C05.Rsp
       command: "RecallScene"
-      groupId: 0x0001
+      groupId: 0x101
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_2_3.yaml:545]: Shouldn't this be using G1 instead of hard-coding the value?  Same for a bunch of other places here...

**Review by bzbarsky-apple — CHANGES_REQUESTED** [review]
This feels like it should be 3 PRs:

1. Fix the code bits that are broken.
2. Update the tests to match the test plan (not mixed in with code changes, so it can get reviewed very carefully).
3. Enable the tests.

File: src/app/tests/suites/certification/Test_TC_S_1_1.yaml [hunk 7]
```diff
@@ -151,7 +151,7 @@ tests:
       response:
           constraints:
               type: list
-              contains: [0, 1, 2, 3, 4, 5, 6, 40, 41, 42]
+              contains: [0, 1, 2, 3, 4, 5, 6, 64, 65, 66]
```
[the remark this answers is not in the document]
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_1_1.yaml:154]: Yes, I could have swore that these commands were all tied together because they all had the same time resolution of a 10th of a second, but after reviewing the spec, it doesn't seem to be the case.

Still, should we at least tie EnhancedAdd and EnhancedView?

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 3 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:183]: Yes but the test plan would have the AddScene overwrite the EnhancedScene, so I assumed this would make sense to ignore. Not that it makes a difference in the expected outcome.

I can add the check on S.S.C00.R but isn't this step redundant if S.S.C40.Rsp is true?

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 6 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:545]: As per above, I can add a check for S.S.C01.R, but executing the two tests seems redundant as they write the same scene to memory and will overwrite each other.

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 5 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:391]: As per above, I can add a check for S.S.C01.R, but executing the two tests seems redundant as they write the same scene to memory and will overwrite each other.

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 4 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml]: 257

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 6 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:525]: It should, problem is I couldn't figure out how to get it to accept a variable.
Is this supported?

File: src/app/tests/suites/certification/Test_TC_S_2_2.yaml (hunk 1 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_2.yaml]: Empty error is the equivalent of a check for success, if any error show up, the test will fail because this expects none.

File: src/app/clusters/level-control/level-control.cpp [hunk 8]
```diff
@@ -918,6 +910,14 @@ static Status moveToLevelHandler(EndpointId endpoint, CommandId commandId, uint8
 
     state->callbackSchedule.runTime = System::Clock::Milliseconds32(0);
 
+#ifdef EMBER_AF_PLUGIN_SCENES
+    // If the level has changed, so the scene is no longer valid.
```
[the remark this answers is not in the document]
Comment (lpbeliveau-silabs) [reply src/app/clusters/level-control/level-control.cpp]: This will be addressed here https://github.com/project-chip/connectedhomeip/pull/29598

File: src/app/tests/suites/certification/Test_TC_S_2_2.yaml (hunk 1 above)
Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_S_2_2.yaml]: The test will fail on error if there is no `response:` section, no?

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 3 above)
Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:178]: I think that if the test plan is wrong we should:

1. Raise a test plan issue.
2. Add a comment here pointing to that test plan issue.

I agree the test plan looks pretty weird to me.

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 5 above)
Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:376]: What this needs is a test plan issue, not unilateral changes to the official certification tests to make them not match the test plan...

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 6 above)
Comment (bzbarsky-apple) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:525]: Ah, this is groupId for the command, not a payload value?  In that case it might not work, in which case we should:

1. Hardcode for now.
2. File an issue on the runner to support it.

File: src/app/tests/suites/certification/Test_TC_S_2_2.yaml (hunk 1 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_2.yaml]: Indeed

Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_2.yaml]: Removed it.

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 3 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml]: Issue [here](https://github.com/CHIP-Specifications/chip-test-plans/issues/3509), comment added

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 6 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml:530]: Issue [here](https://github.com/project-chip/connectedhomeip/issues/29637), commend added.

File: src/app/tests/suites/certification/Test_TC_S_2_3.yaml (hunk 5 above)
Comment (lpbeliveau-silabs) [reply src/app/tests/suites/certification/Test_TC_S_2_3.yaml]: Issue [here](https://github.com/CHIP-Specifications/chip-test-plans/issues/3509), comment added

**Review by jmartinez-silabs — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_S_1_1.yaml [hunk 9]
```diff
@@ -60,80 +60,59 @@ tests:
...
       command: "readAttribute"
       attribute: "AttributeList"
       response:
           constraints:
               type: list
-              contains: [0, 1, 2, 3, 4, 6, 7, 65528, 65529, 65531, 65532, 65533]
+              contains: [6]
+
+    - label:
+          "Step 4e: Read optionnal attribute(FabricSceneInfo) in AttributeList
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_S_1_1.yaml:107]:
```suggestion
          "Step 4e: Read optional attribute(FabricSceneInfo) in AttributeList
```

**Review by bzbarsky-apple — APPROVED** [review]

**Review by rcasallas-silabs — APPROVED** [review]
